### PR TITLE
Add Ideogram 4 local prompt expansion support

### PR DESCRIPTION
**Pull request by omercelik**
## Summary

Builds on the Ideogram 4 base-model support merged in #1281 and adds optional local prompt expansion for ordinary text prompts.

This PR adds:

- `prompt_expansion_model` support for CLI, Python image-generation requests, and the OpenAI-compatible image-generation API
- Schema-constrained Ideogram 4 JSON caption generation using an existing mlx-vlm text/VLM model
- Plain-prompt wrapping, structured-caption validation, fallback behavior, and `revised_prompt` metadata
- Ideogram 4 prompting documentation and focused test coverage

## Prompting behavior

Ideogram 4 was trained on structured JSON captions. The prompt path supports three workflows:

1. Structured JSON captions are passed through unchanged.
2. Ordinary text prompts are wrapped into a minimal JSON caption by default.
3. Ordinary text prompts can be expanded into a richer structured caption by a local mlx-vlm text/VLM model before image generation.

The expansion model uses `build_json_schema_logits_processor` with an Ideogram 4 caption schema. The prompt instructs the model to preserve requested visible text verbatim, use concrete scene descriptions, use one object element per named subject, and use normalized bounding boxes only when useful.

This is a local mlx-vlm prompt-expansion path. It does not call the hosted Ideogram Magic Prompt service.

## Usage

CLI:

```sh
python -m mlx_vlm generate_image \
  --model ideogram-ai/ideogram-4-fp8 \
  --prompt "A cinematic photo of a glass teapot on a rainy London cafe table" \
  --prompt-expansion-model mlx-community/gemma-4-12B-it-4bit \
  --size 1024x1024 \
  --output outputs/ideogram4-expanded.png
```

Python:

```python
request = ImageGenerationRequest(
    prompt="A cinematic photo of a glass teapot on a rainy London cafe table",
    width=1024,
    height=1024,
    extra={
        "prompt_expansion_model": "mlx-community/gemma-4-12B-it-4bit",
    },
)
```

The OpenAI-compatible `/v1/images/generations` request accepts the optional `auto_json_caption` and `prompt_expansion_model` fields. Wrapped or expanded captions are returned as `revised_prompt`.

## Failure behavior

- JSON object prompts skip prompt expansion.
- Invalid expansion output falls back to the minimal JSON caption wrapper unless `auto_json_caption=False` is set.
- Model loading and inference errors are raised so configuration problems are not hidden.

## Testing

```sh
python -m pytest mlx_vlm/tests/test_ideogram4.py mlx_vlm/tests/test_flux2.py mlx_vlm/tests/test_generate.py mlx_vlm/tests/test_server.py -q
# 254 passed

pre-commit run --files <changed files>
# black, isort, autoflake passed
```

A live offline smoke test also passed using cached `ideogram-ai/ideogram-4-fp8` and `mlx-community/gemma-4-12B-it-4bit` models with a 256x256 image and 10 inference steps.

### CODE REVIEW

**Comment by lucasnewman** [comment]
@omercelik This is very cool -- we merged support for the base model [here](https://github.com/Blaizzy/mlx-vlm/pull/1281). Would you mind updating your implementation to use that instead? Can we also call the option `--prompt-expansion-model` so it's a bit more literal about what it will do?

**Comment by omercelik** [comment]
@lucasnewman sure, updated the PR.

**Review by lucasnewman — APPROVED** [review]
Awesome!
